charybdis/insn.py

```diff
@@ -96,7 +96,13 @@ class R16(enum.Enum):
     SP = "SP"
 
 
-InsnOperand = Union[Label, R8, R16, U8, U16]
+@dataclasses.dataclass
+class Indirect:
+    """Indirect addressing mode via 16-bit register"""
+    reg: R16
```

Comment:
This is excellent &mdash; exactly what I had in mind when it came to expanding `InsnOperand`.

However, the name should probably be adjusted due to the fact that there are multiple indirect addressing modes, eg: `LD A, [$FF00 + n]`, `LD A, [$FF00 + C]`.

What about `IndirectR16`? The suffix suggests further types `IndirectU8`, `IndirectC`, et al.

Reply:
Per offline conversation: will be addressed in next PR.